CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/FileInputPane.java

```diff
@@ -292,21 +293,26 @@ private FileChooserBuilder getFileChooser(final PluginParameter<FileParameterVal
                 .setAcceptAllFileFilterUsed(extensionFilter == null || FileParameterType.isAcceptAllFileFilterUsed(parameter));
 
         if (extensionFilter != null) {
-            for (final String extension : extensionFilter.getExtensions()) {
-                // Add a file filter for all registered exportable file types.
-                fileChooserBuilder = fileChooserBuilder.addFileFilter(new FileFilter() {
-                    @Override
-                    public boolean accept(final File file) {
-                        final String name = file.getName();
-                        return (file.isFile() && StringUtils.endsWithIgnoreCase(name, extension)) || file.isDirectory();
+            // Add a file filter for all registered exportable file types.
+            List<String> extensions = extensionFilter.getExtensions();
+            
+            fileChooserBuilder = fileChooserBuilder.addFileFilter(new FileFilter() {
+                @Override
+                public boolean accept(final File file) {
+                    final String name = file.getName();
+                    String ext = "";
+                    int dotIndex = name.lastIndexOf('.');
```

Comment:
final. `ext` could also be made final by converting that declaration to a ternary statement